contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinNerve.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+
+contract MixinNerve {
+
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibSafeMathV06 for uint256;
+    using LibRichErrorsV06 for bytes;
+
+
+    struct NerveBridgeData {
+        address pool;
+        bytes4 exchangeFunctionSelector;
+        int128 fromCoinIdx;
+        int128 toCoinIdx;
+    }
+
+    function _tradeNerve(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        internal
+        returns (uint256 boughtAmount)
+    {
+        // Basically a Curve fork but the swap option has a deadline
+
+        // Decode the bridge data to get the Curve metadata.
+        NerveBridgeData memory data = abi.decode(bridgeData, (NerveBridgeData));
+        sellToken.approveIfBelow(data.pool, sellAmount);
+        uint256 beforeBalance = buyToken.balanceOf(address(this));
```

Comment:
Can we just trust the `dy` returned by `swap()`?

Reply:
Oh woops missed that early guy, yeah I just trust the returned value, let me rm this line